packages/neon/neon/lib/src/models/account.dart

```diff
@@ -72,10 +72,24 @@ class Account implements Credentials {
     return _idCache[key] = sha1.convert(utf8.encode(key)).toString();
   }
 
+  /// A human readable representation of [username] and [serverURL].
   String get humanReadableID {
-    final uri = Uri.parse(serverURL);
-    // Maybe also show path if it is not '/' ?
-    return '$username@${uri.port != 443 ? '${uri.host}:${uri.port}' : uri.host}';
+    final buffer = StringBuffer()
+      ..write(username)
+      ..write('@')
+      ..write(serverURL.host);
+
+    if (serverURL.hasPort) {
+      buffer
+        ..write(':')
+        ..write(serverURL.host);
+    }
+
+    if (!serverURL.hasEmptyPath) {
```

Comment:
How does this handle `/` as path?

Reply:
We already assert in the dynamite client that the path should not end with a `/`

Reply:
https://github.com/nextcloud/neon/pull/603/files#diff-8dde2cc07e164437d15007e4dfc0d39721404c1370d1b4008087877d6f0d9c83R120

Reply:
Hm, now I wonder how the UI will handle it if some puts in a URL with a trailing slash on the login screen

Reply:
We allow trailing slashes and manually strip it before passing it into the bloc.

I'll need to check for the loginQrCode though

Reply:
We also only assert this meaning that it will get compiled away for production.

We might want to throw an exception

Reply:
@provokateurin what do you think?

Reply:
In our login logic yes, but I wouldn't do it in the nextcloud package.